### PR TITLE
On core changes for secure erase progress

### PR DESCRIPTION
**Pull request by pengz1**
Changes includes:
1. Create progress property in taskdependencies, graphObject and its related methods
2. Create progress related events

### CODE REVIEW

**Comment by pengz1** [comment]
@RackHD/corecommitters  @cgx027 @iceiilin

**Comment by rackhdmirror** [comment]
@RackHD/corecommitters please review.

File: lib/workflow/stores/mongo.js [hunk 1]
```diff
@@ -985,5 +985,82 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _, san
...
+            instanceId: data.graphId,
+            _status: {$in: Constants.Task.ActiveStates}
+        };
+        var options = {
+            multi: true
+        };
+        return waterline.graphobjects.findOne({instanceId: data.graphId})
+        .then(function(graphObject){
+            if (!_.has(data, 'graphName')) {
+                data.graphName = graphObject.definition.friendlyName;
```
Comment (yyscamper) [inline lib/workflow/stores/mongo.js]: I don't see anywhere in this function will use the `graphName`, so why do this change?

File: lib/workflow/stores/mongo.js [hunk 2]
```diff
@@ -985,5 +985,82 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _, san
...
+            }
+            //graph progres percentage might not change when task prrogress
```
Comment (yyscamper) [inline lib/workflow/stores/mongo.js]: typo: prrogress

File: spec/lib/protocol/events-spec.js [hunk 3]
```diff
@@ -478,4 +478,29 @@ describe("Event protocol subscribers", function () {
             });
         });
     });
+    
+    describe("publish graph progress event", function () {
+        it("should publish graph progress event", function () {
+            var uuid = helper.injector.get('uuid');
+            var data = {
+                    graphId: uuid.v4(),
```
Comment (yyscamper) [inline spec/lib/protocol/events-spec.js]: indent error

File: spec/lib/workflow/messengers/messenger-amqp-spec.js [hunk 4]
```diff
@@ -140,6 +141,25 @@ describe('Task/TaskGraph AMQP messenger plugin', function () {
         });
     });
 
+    it('should wrap event protocol publishProgressEvent method', function(){
+        var uuid = helper.injector.get('uuid'); 
+        var data = {
+                graphId: uuid.v4(),
```
Comment (yyscamper) [inline spec/lib/workflow/messengers/messenger-amqp-spec.js]: indent error

File: lib/workflow/stores/mongo.js (hunk 1 above)
Comment (pengz1) [inline lib/workflow/stores/mongo.js]: Just to align with taskProgress. Both task and workflow progress will includes task/workflow name in AMQP message.

**Comment by pengz1** [comment]
test this please

**Pull request merged**